compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/optimizations/PropertyAccessorInlineLowering.kt

```diff
@@ -46,13 +48,25 @@ class PropertyAccessorInlineLowering(private val context: CommonBackendContext)
             }
             if (property.isEffectivelyExternal()) return expression
 
+            val backingField = property.backingField ?: return expression
+
             if (property.isConst) {
                 val initializer =
-                    (property.backingField?.initializer ?: error("Constant property has to have a backing field with initializer"))
-                return initializer.expression.deepCopyWithSymbols()
+                    (backingField.initializer ?: error("Constant property has to have a backing field with initializer"))
+                val constExpression = initializer.expression.deepCopyWithSymbols()
+                val receiver = expression.dispatchReceiver
+                if (receiver != null && receiver is IrCall) {
```

Comment:
I think instead of `receiver is IrCall` I'd rather go with `!receiver.isPure()`

Reply:
Do you mean `isPure` from `org.jetbrains.kotlin.ir.backend.js.utils`? It has parameters and then it should be replaced.
Also I found `isPure` function, but it's `private` and inside K/N, so also need replacement.

Reply:
I mean the first one. What the problem with parameters? I think `!receiver.isPure(true)` would be enough.